opentelemetry/src/logs/record.rs

```diff
@@ -1,79 +1,46 @@
 use crate::{
-    trace::{SpanContext, SpanId, TraceContextExt, TraceFlags, TraceId},
+    trace::{SpanContext, TraceContextExt},
     Array, Key, StringValue, Value,
 };
 use std::{borrow::Cow, collections::HashMap, time::SystemTime};
 
-#[derive(Debug, Clone)]
-#[non_exhaustive]
-/// LogRecord represents all data carried by a log record, and
-/// is provided to `LogExporter`s as input.
-pub struct LogRecord {
-    /// Event name. Optional as not all the logging API support it.
-    pub event_name: Option<Cow<'static, str>>,
+/// Abstract interface for managing log records. The implementation is provided by the SDK
+/// Abstract interface for log records in an observability framework.
+pub trait LogRecord {
+    /// Sets the creation timestamp.
+    fn set_timestamp(&mut self, timestamp: SystemTime);
 
-    /// Record timestamp
-    pub timestamp: Option<SystemTime>,
+    /// Sets the observed event timestamp.
+    fn set_observed_timestamp(&mut self, timestamp: SystemTime);
 
-    /// Timestamp for when the record was observed by OpenTelemetry
-    pub observed_timestamp: SystemTime,
+    /// Associates a span context.
+    fn set_span_context(&mut self, context: &SpanContext);
```

Comment:
I am totally okay to remove this method. As of today, SDK sets this automatically. For now, we can rely on that, and later add this API, if needed.

Reply:
I think we should remove this, as SDK currently overwrites this. Better to not offer this API now, and wait for the tokio-tracing discussions to finish before exposing this.

Reply:
Makes sense, it's optional as per the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md#logger-operations) too:

``
The [Context](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/README.md) associated with the LogRecord. The API MAY implicitly use the current Context as a default behavior.
``

Reply:
Just to confirm, you mean both `set_span_context` and `set_context` ?

Reply:
Ok removed both for now :)